tls/s2n_cipher_suites.c

```diff
@@ -1168,27 +1171,57 @@ static int s2n_wire_ciphers_contain(const uint8_t *match, const uint8_t *wire, u
     return 0;
 }
 
+bool s2n_cipher_suite_uses_chacha20_alg(struct s2n_cipher_suite *cipher_suite) {
+    return cipher_suite && cipher_suite->record_alg && cipher_suite->record_alg->cipher == &s2n_chacha20_poly1305;
+}
+
+static S2N_RESULT s2n_client_signalled_chacha20_boosting(struct s2n_stuffer wire_stuffer, uint32_t cipher_suite_len, bool *client_signalled_chacha20_boosting) {
```

Comment:
Do we actually care if this method fails vs returns false? If there was theoretically a bug in the chacha20 boosting logic and it sometimes errored, should that affect the non-chacha20 path? You might not need to separate error from false here.

Reply:
This function does three things:

1. reads from the stuffer the first client iana
2. retrieves the associated cipher suite from iana
3. checks if the cipher suite uses chacha20_poly1305

I agree that if we error in any of those steps then we can just return false and avoid the chacha20 path.  Does this mean this should just return a bool and avoid all the `GUARD` checks?

Reply:
I would do the opposite. GUARD checks make everything cleaner to read. Just check if the method succeeds, and name it something that indicates that.